src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -156,7 +156,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	get resource(): URI {
 		return URI.from({
 			scheme: Schemas.vscodeTerminal,
-			path: `/${this.instanceId}`,
+			path: `/${this.persistentProcessId}`,
```

Comment:
Hmm, pretty sure this will break things within the same window. But it is the more accurate thing to use when dragging across windows. This also brings up the question of what happens when dragging terminals we cannot transfer to another window such as:

- Tasks - we probably want to avoid doing this as the task logic won't transfer too
- Extension `Pseudoterminal` terminals - these are bound to the window's extension host

I can see us solving this in a couple of ways:

1. Use persistent process ID everywhere instead of instance ID, I think we'll have problems here because some terminals aren't hosted on the ptyhost but in the extension host instead.
2. Include the persistent process ID as maybe part of the query string of the URI, this seems kind of messy
3. Change all URIs and their handling to use the format `/windowId/instanceId`, another benefit of this is we could request the specific window to detach.

3 seems like the way to go.